vignettes/col_counts.Rmd

```diff
@@ -0,0 +1,88 @@
+  %\VignetteEngine{knitr::rmarkdown}
+editor_options: 
+  chunk_output_type: console
+---
+
+```{r, echo=FALSE}
+knitr::opts_chunk$set(comment = "#")
+```
+
+# Column Counts and Formats
+
+Main drivers of column counts are parameters `show_colcounts` and `colcount_format` in `split_cols_by()`. The former is a boolean that determines whether to show the column counts in the table, while the latter is a string that specifies the format of the column counts. The default value for `show_colcounts` is `FALSE`, while the default value for `colcount_format` is `"(N=xx)"`.
+
+```{r}
+lyt <- basic_table() %>%
+  split_cols_by("ARMCD", show_colcounts = TRUE, colcount_format = "N=xx") %>%
+  split_cols_by("STRATA2", show_colcounts = TRUE) %>%
+  split_cols_by("STRATA1") %>%
+  add_overall_col("All") %>%
+  analyze("AGE", afun = max, format = "xx.x")
+
+tbl <- build_table(lyt, ex_adsl)
+tbl
+```
+
+We can control the column counts with the following functions:
+
+```{r}
+col_counts(tbl) # vector of all of them (visible and not visible) -> difficult to read/use for nested structures
+coldf <- make_col_df(tbl) # to check the structure of the columns as a data.frame
+coltree_structure(tbl) # Prints the structure of the columns as a tree
+```
+
+`make_col_df` is a fundamental function that is used also internally to keep track of the tree structure of the columns. It returns a `data.frame` very similar to `make_row_df`, but for columns. The `coltree_structure` function is a wrapper around `make_col_df` that prints the structure of the columns as a tree.
+
+To retrieve the column counts for a specific column, use pathing! Also visibility can be accessed and modified with the following functions:
+
+```{r}
+facet_colcount(tbl, coldf$path[[1]]) # To get the column counts for a specific column at a specific level
+colcount_visible(tbl, coldf$path[[1]]) # not visible!!! # FALSE
+facet_colcount(tbl, coldf$path[[1]][c(1, 2)]) # visible
+colcount_visible(tbl, coldf$path[[1]][c(1, 2)]) # TRUE
+```
+
+Formats can differ as we have shown in the example above. We can change the format of the column counts with the following function:
+
+```{r, eval=FALSE}
+colcount_format(tbl) <- "[N<=xx]" # Works only with the default format (last line)
+colcount_visible(tbl, coldf$path[[1]]) <- TRUE
+tbl
+# Error in h(simpleError(msg, call)) :
+#   error in evaluating the argument 'x' in selecting a method for function 'toString': Detected different colcount visibility among sibling facets (those arising from the same split_cols_by* layout instruction). This is not supported.
+# Set count values to NA if you want a blank space to appear as the displayed count for particular facets.
+# First disagreement occured at paths:
+# ARMCD[ARM A]->STRATA2[S1]->STRATA1[A]
+# ARMCD[ARM A]->STRATA2[S1]->STRATA1[B]
+```
+
+For removing the column counts from a specific column, we can do it in two ways:
+
+```{r}
+ccnts <- col_counts(tbl)
+ccnts[1] <- NA_integer_
+col_counts(tbl) <- ccnts
+tbl
+```
```

Comment:
This part is still to be completed with the two cases you showed us on Wednesday (I have to go OoO, we can catch up on this another day)